python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -529,99 +519,56 @@ def asset_key_for_input(self, input_name: str) -> AssetKey:
             return key
 
     @public
-    def asset_partition_key_for_output(self, output_name: str = "result") -> str:
-        """Returns the asset partition key for the given output. Defaults to "result", which is the
-        name of the default output.
-        """
-        return self._step_execution_context.asset_partition_key_for_output(output_name)
+    def partition_key_for_asset(
+        self, asset: CoercibleToAssetKey, is_dependency: bool = False
+    ) -> Union[str, MultiPartitionKey]:
+        """Returns the partition key of provided asset."""
+        return self._step_execution_context.asset_partition_key_for_asset(
+            asset, is_dependency=is_dependency
+        )
 
     @public
-    def asset_partitions_time_window_for_output(self, output_name: str = "result") -> TimeWindow:
-        """The time window for the partitions of the output asset.
+    def partitions_time_window_for_asset(
```

Comment:
kinda odd to me that this one is plural `partitions`. Any objection to renaming `partition_time_window_for_asset`?

Reply:
Need to keep the old one around and deprecate it

Reply:
the old one being `asset_partitions_time_window_for_output`? I don't think we should rename that, and it's been moved to the bottom of the file and marked deprecated.

Reply:
ah I misread what was going on 👍🏻

Reply:
yeah moving the methods around in the file made the diff really annoying. I'll stack a PRT before this one that just moves the methods to clean things up

Reply:
ok stacked a temp PR before this one that should hopefully make the diff a bit better to read. It's still pretty beefy, but at least GitHub is showing the changes in a way that's truer to what the PR is doing